lattices/src/lib.rs

```diff
@@ -107,3 +107,18 @@ pub trait IsTop {
     /// Returns if `self` is lattice top (⊤).
     fn is_top(&self) -> bool;
 }
+
+/// Trait to atomize a lattice into individual elements. For example, a [`set_union::SetUnion`]
+/// will be broken up into individual singleton elements.
+///
+/// Formally, breaks up `Self` into an set of lattice points forming a (strong) [antichain](https://en.wikipedia.org/wiki/Antichain).
+pub trait Atomize {
+    /// The type of atoms for this lattice.
+    type Atom: 'static;
+
```

Comment:
Looks like the output of atomize need not be a lattice point from the domain of the input, which is potentially problematic.

Reply:
Added bound `Atomize: Merge<Self::Atom>`